expression/generator/control_vec.go

```diff
@@ -238,30 +382,39 @@ func (b *builtinIf{{ .TypeName }}Sig) vecEval{{ .TypeName }}(input *chunk.Chunk,
 	if err := b.args[0].VecEvalInt(b.ctx, input, buf0); err != nil {
 		return err
 	}
-
+	sc := b.ctx.GetSessionVars().StmtCtx
+	beforeWarns := sc.WarningCount()
 {{- if .Fixed }}
-	if err := b.args[1].VecEval{{ .TypeName }}(b.ctx, input, result); err != nil {
-		return err
-	}
+	err = b.args[1].VecEval{{ .TypeName }}(b.ctx, input, result)
 {{- else }}
 	buf1, err := b.bufAllocator.get(types.ET{{ .ETName }}, n)
 	if err != nil {
 		return err
 	}
 	defer b.bufAllocator.put(buf1)
-	if err := b.args[1].VecEval{{ .TypeName }}(b.ctx, input, buf1); err != nil {
-		return err
-	}
+	err = b.args[1].VecEval{{ .TypeName }}(b.ctx, input, buf1)
 {{- end }}
+	afterWarns := sc.WarningCount()
+	if err != nil || afterWarns > beforeWarns {
```

Comment:
Is it necessary to check the error here? Are warnings enough?

Reply:
Some warnings may become errors in different sql mode. So each error and warning should detected here.